### ISSUE TITLE
New maintainer for pydap?

### ISSUE THREAD

**Issue by shoyer** [pydap/pydap#14]
@robertodealmeida -- Many of us still find pydap to be quite useful, but it's clear that you no longer have the time or inclination to support it. That's OK -- we all need to move on eventually -- but would you be willing to add someone else as an administrator who can merge pull requests and issue releases on pypi? In particular, it would be extremely valuable if someone is able to merge in Python 3 support.

If you're not willing to add another maintainer, then I will probably fork the project. But I would rather avoid such a split.

@jameshiebert @SiggyF @jgallagher59701 @jblarsen  @jfburkhart  -- All of you have outstanding pull requests. Are any of you interested in being a maintainer for this project? If need be, I will step up, but I have way too many open source obligations already...

**Comment by bolhovsky** [comment]
Thanks @shoyer for raising the question. And +1 for Python 3 support: we've been waiting for this for a long time.

**Comment by robertodealmeida** [comment]
Hi, guys!

Sure, I would appreciate if anybody could take over, specially because @jameshiebert did all the work to support Python 3 and the PR has been waiting for my review for over a year now. I learned a lot developing pydap, but I don't have the bandwidth to support a project that I don't use anymore, and after 12 years my interested has switched elsewhere.

I'll add @shoyer as a collaborator in this repo, and can add more people. Also, if you could send me your PyPI usernames I can edit the package information so other people can upload.

Thanks,
--Roberto

**Comment by shoyer** [comment]
Great, thanks Roberto! I'm Stephan.Hoyer on PyPI.

On Thu, Oct 1, 2015 at 3:08 PM, Roberto De Almeida <notifications@github.com

**Comment by marcelorodriguesss** [comment]
Hi Roberto,

My name is Marcelo and I work at Funceme (Fortaleza - Ceará - Brasil). I
use Pydap and Pupynere everyday. I am new in python programming, there is
some way to contribute in Pydap project?

Marcelo Rodrigues
Em 01/10/2015 19:08, "Roberto De Almeida" <notifications@github.com>
escreveu:

**Comment by jameshiebert** [comment]
Hi All,
Thanks for bringing this up @shoyer! Personally, I don't think that I have the capacity to serve as the maintainer, but I'm happy to continue contributing to the project (through further PRs and such).

**Comment by shoyer** [comment]
I don't have any plans for this project beyond the occasional bug fix and Python 3 support, so if anyone else has bigger ideas, they should definitely step up.

Fortunately, Roberto has left pydap in pretty good shape: the project is generally quite mature. It looks like there's been a lot of work on `develop` branch that hasn't made it into a release yet. @robertodealmeida was that mostly done and you were only waiting on adding Python 3 before making a release? If not, it would be good to know where things stand.

We do seem to have automated testing running (via Travis-CI -- I just added a link to the readme), but it doesn't seem to have run on the last few PRs. @robertodealmeida could you take a quick look and verify that there's still a Travis-CI hook in this repo's settings? The other option is transfer this repo to a pydap organization on GitHub and add me as a maintainer -- that way I can edit all the hooks.

**Comment by jfburkhart** [comment]
@shoyer and @robertodealmeida , great that you guys will move this package toward Python 3. I will certainly be willing to contribute in the form of PRs, but would fear I would not be a well engaged maintainer (as noted by my +9 day response due to having been in the field!).

**Comment by jgallagher59701** [comment]
> On Oct 12, 2015, at 3:49 PM, John F. Burkhart <notifications@github.com> wrote:
> 
> @shoyer <https://github.com/shoyer> and @robertodealmeida <https://github.com/robertodealmeida> , great that you guys will move this package toward Python 3. I will certainly be willing to contribute in the form of PRs, but would fear I would not be a well engaged maintainer (as noted by my +9 day response due to having been in the field!).
> 
> 
1+ WRT moving toward Python 3

I also cannot take this on - I’m not that good at Python nor do I have much time. However, ROberto and I have talked about adding DAP4 support to pydap - I can help with that, modulo my lack of much Python skill.
> —
> Reply to this email directly or view it on GitHub <https://github.com/robertodealmeida/pydap/issues/14#issuecomment-147531088>.
> 

--
James Gallagher
jgallagher@opendap.org

**Comment by jameshiebert** [comment]
Is there still any interest in this from folks? My organization is going to be needing to update our data portal(s) this year and I'd be willing to pick up the torch for a little while. It's been a couple years since I've worked with the code in great detail, but we're kind of tired of running our silly forked version, and I think that it will come back to me.

**Comment by shoyer** [comment]
I think it would be great if you could take over maintenance. I had an
abstract interest in Python 3 support so we could move to Python 3 at my
old job, but I've moved on now.

**Comment by MattF-NSIDC** [comment]
+1

@jameshiebert I am absolutely interested in seeing this project maintained, especially python 3 support. I was having a conversation with co-workers about this at the exact moment you posted that question.

**Comment by jgallagher59701** [comment]
+1

A big thumbs up from OPeNDAP.


> On May 4, 2016, at 16:02, MattF-NSIDC <notifications@github.com> wrote:
> 
> +1
> 
> @jameshiebert <https://github.com/jameshiebert> I am absolutely interested in seeing this project maintained, especially python 3 support. I was having a conversation with co-workers about this at the exact moment you posted that question.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/robertodealmeida/pydap/issues/14#issuecomment-217017490>

--
James Gallagher
jgallagher@opendap.org

**Comment by jameshiebert** [comment]
Great, that sounds like some enthusiastic support from respectable people. I'll start working on getting the Python 3 stuff merged into the 3.2 dev branch and get that release ready to go. @shoyer and @robertodealmeida are there any credentials that you can pass along that I'll need as a maintainer?

**Comment by shoyer** [comment]
@robertodealmeida gave me push access to this repo so I can merge PRs, but I can't add anyone else. I do think I can add you on PyPI though.

**Comment by ghisvail** [comment]
Wouldn't it make sense to transfer the repository to an organizational account where you guys can share the maintenance of the project?

**Comment by robertodealmeida** [comment]
I can add more maintainer to the repo, or if you want to use a different repo I can add a note here and in pydap.org pointing to it.

**Comment by robertodealmeida** [comment]
I can also transfer the ownership to @shoyer or @jameshiebert.

**Comment by shoyer** [comment]
It looks like github.com/pydap is available.

I would recommend either staying here or transfer this repo rather than creating a new one, though.

**Comment by ghisvail** [comment]
> transfer this repo rather than creating a new one, though.

You guys could transfer to pydap/pydap.

**Comment by robertodealmeida** [comment]
How about https://github.com/opendap? @jgallagher59701, would that be fine?

**Comment by jgallagher59701** [comment]
> On May 5, 2016, at 09:24, Roberto De Almeida <notifications@github.com> wrote:
> 
> How about https://github.com/opendap <https://github.com/opendap>? @jgallagher59701 <https://github.com/jgallagher59701>, would that be fine?
> 
> 
Yes, that would be fine.

James

> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/robertodealmeida/pydap/issues/14#issuecomment-217184776>

--
James Gallagher
jgallagher@opendap.org

**Comment by jameshiebert** [comment]
I like @ghisvail's idea of transferring ownership to a new organization, and if `pydap` is available and @jgallagher59701 prefers that over `opendap`, then let's go with that. I'll create github.com/pydap and then @robertodealmeida can you transfer ownership of pydap and the related namespace repos (pydap.handers.\*, pydap.repsonses.\*)? @shoyer if you want to add me on PyPI, I'm 'jamesmhiebert'.

**Comment by jameshiebert** [comment]
> I'll create github.com/pydap

Done. If anyone further is interested in being a member of the pydap organization, please let me know.

**Comment by jgallagher59701** [comment]
> On May 5, 2016, at 11:43, James Hiebert <notifications@github.com> wrote:
> 
> I like @ghisvail <https://github.com/ghisvail>'s idea of transferring ownership to a new organization, and if pydap is available and @jgallagher59701 <https://github.com/jgallagher59701> prefers that over opendap, then let's go with that. I'll create github.com/pydap and then @robertodealmeida <https://github.com/robertodealmeida> can you transfer ownership of pydap and the related namespace repos (pydap.handers.*, pydap.repsonses.*)? @shoyer <https://github.com/shoyer> if you want to add me on PyPI, I'm 'jamesmhiebert'.
> 
> 

Good! I’d like to maintain pydap’s independence, that’s why I prefer having pydap not come under the umbrella of an existing org.

I’d like to be a member of pydap.

Thanks,
James
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/robertodealmeida/pydap/issues/14#issuecomment-217222649>

--
James Gallagher
jgallagher@opendap.org

**Comment by robertodealmeida** [comment]
I'm not sure I can transfer ownership of the repos to github.com/pydap — my understanding is that the repos need to be forked there, no?

**Comment by shoyer** [comment]
If you go to Settings for this repo and scroll down to the bottom of the
first page you should see an option for "Transfer ownership".

On Fri, May 6, 2016 at 9:24 AM, Roberto De Almeida <notifications@github.com

**Comment by MattF-NSIDC** [comment]
He has to be an admin on the pydap organization as well

**Comment by robertodealmeida** [comment]
Done. :)

One more thing: the current website is hosted in an S3 bucket, and updated using Paver, but of course it depends on my keys to do the upload. Let me know if you want to point the website to some other host, or take over the bucket.

**Comment by shoyer** [comment]
Normally, I would recommend using ReadTheDocs, which rebuilds Sphinx documentation automatically.  But it doesn't seem to know how to handle the lack of a setup.py file in the source distribution: https://readthedocs.org/projects/pydap/

Another low maintenance option is GitHub pages: https://pages.github.com/

**Comment by ghisvail** [comment]
:+1:  for either RTD or ghpages. RTD is perhaps the expected standard for Python packages.

**Comment by robertodealmeida** [comment]
One thing about the current repo: the develop branch of Pydap was practically ready for a 3.2 release. If I remember correctly it had 100% unit test coverage, and @jameshiebert submitted a PR adding Python 3 support.

**Comment by jameshiebert** [comment]
Hi all. Thanks for initiating this discussion and for all the help in setting me up as the new maintainer. I think that I'm good to go for proceeding with development and working towards the 3.2 release. I'm going to close this issue, but I'll open another one to resolve the (somewhat separate) issue of where to host the docs.

**Comment by shoyer** [comment]
Maybe @laliberte would be a good addition to the pydap team?

**Comment by laliberte** [comment]
I could definitely spend some time looking at client-related issues and PRs.
